library/common/http/client.cc

```diff
@@ -99,14 +108,52 @@ void Client::DirectStreamCallbacks::encodeData(Buffer::Instance& data, bool end_
     return;
   }
 
```

Comment:
How would you feel about this?

```suggestion
if (!explicit_buffering_) {
  bytes_to_send_ = data.length();
}

```

Then we could always rely on bytes_to_send_, we could always do a simple `std::min` before sending, and you wouldn't need to treat 0 as a special value (which feels a little bit risky to me, even though I see where it's guarded).

Reply:
I think I'm maybe missing something. Why would this introduce an upcall for cronvoy or introduce additional context switches? I'm just suggesting adding these lines so that later we don't need extra checks on `bytes_to_send_` - we can rely on it being the correct value regardless of whether `explicit_buffering_` is enabled.